src/main/java/com/truelayer/java/paymentsproviders/PaymentsProvidersHandler.java

```diff
@@ -1,26 +1,40 @@
 package com.truelayer.java.paymentsproviders;
 
+import static com.truelayer.java.Constants.Scopes.PAYMENTS;
+
+import com.truelayer.java.IAuthenticatedHandler;
+import com.truelayer.java.entities.RequestScopes;
 import com.truelayer.java.http.entities.ApiResponse;
 import com.truelayer.java.paymentsproviders.entities.PaymentsProvider;
 import java.util.concurrent.CompletableFuture;
-import lombok.Value;
+import lombok.Builder;
+
 
 /**
  * {@inheritDoc}
  */
-@Value
-public class PaymentsProvidersHandler implements IPaymentsProvidersHandler {
+@Builder
+public class PaymentsProvidersHandler implements IAuthenticatedHandler, IPaymentsProvidersHandler {
 
-    String clientId;
```

Comment:
There seems to be an extra line break here, currently failing the CI. 
See [this](https://github.com/TrueLayer/truelayer-java?tab=readme-ov-file#code-linting) on how to run linting locally.